ZeCamp/ZeCamp/Schedule/ScheduleScreen.swift

```diff
@@ -0,0 +1,136 @@
+        subview.topAnchor.constraint(equalTo: self.topAnchor, constant: 0).isActive = true
+        subview.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: 0).isActive = true
+        subview.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 0).isActive = true
+        subview.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: 0).isActive = true
+    }
+}
+
+struct ScheduleScreen {
+    
+    private let dataSource: UITableViewDataSource!
+    
+    init(schedule: Schedule) {
+        dataSource = ScheduleDataSource(schedule)
+    }
+    
+    func makeViewController() -> UIViewController {
+        let viewController = UIViewController()
+        let scheduleTable = ScheduleTableView()
+
+        scheduleTable.strongDataSource = dataSource
+        scheduleTable.rowHeight = UITableViewAutomaticDimension
+        scheduleTable.estimatedRowHeight = 50
+        
+        viewController.view.addFillingSubview(scheduleTable)
+        
+        return viewController
+    }
+}
+
+class ScheduleTableView: UITableView {
+    fileprivate var strongDataSource: UITableViewDataSource? {
+        didSet {
+            dataSource = strongDataSource
+        }
+    }
+}
+
+struct TimeSlot {
+    let date: Date
+    let events: [Event]
+}
+
+class ScheduleDataSource: NSObject, UITableViewDataSource {
+    
+    let timeSlots: [TimeSlot]
+    
+    init(_ schedule: Schedule) {
+        self.timeSlots = schedule.asOrderedTimeSlots()
+    }
+    
+    @available(iOS 2.0, *)
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return timeSlots.count
+    }
+    
+    @available(iOS 2.0, *)
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return timeSlots[section].events.count
+    }
+    
+    @available(iOS 2.0, *)
+    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return timeSlots[section].sectionTitle
+    }
+    
+    @available(iOS 2.0, *)
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let event = self.timeSlots[indexPath.section].events[indexPath.row]
+        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
+        
+        let stack = UIStackView()
+        stack.axis = .vertical
+        
+        let eventName = UILabel()
+        eventName.text = event.name
+        stack.addArrangedSubview(eventName)
+        
+        let eventDetails = UILabel()
+        eventDetails.text = "\(event.startTime) - \(event.endTime) - \(event.location)"
+        stack.addArrangedSubview(eventDetails)
+        
+        cell.addFillingSubview(stack)
+        
+        return cell
+    }
+}
+
+//TODO: time zone?
+fileprivate let dayAndTimeFormatter: DateFormatter = {
+    let formatter = DateFormatter()
+    formatter.dateFormat = "EEEE, HH:mm"
```

Comment:
Instead of hardcoding the format, you can ask the OS to give you the final format since the preferred format for the same information may vary by locale:
```swift
formatter.dateFormat = DateFormatter.dateFormat(fromTemplate: "EEEEHHmm", options: 0, locale: nil)
```
In this case, dateFormat resolves to `EEEE HH:mm`.